pkg/kv/kvserver/rangefeed_restarter.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package kvserver
+
+import (
+	"context"
+	"math"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/errors"
+)
+
+var rangefeedRestartInterval = settings.RegisterDurationSetting(
+	settings.SystemOnly,
+	"kv.rangefeed.restart_interval",
+	"minimum interval between rangefeed restarts to conform to new settings",
+	50*time.Millisecond,
```

Comment:
This works out to ~40 minutes for 50k replicas, which seems reasonable. We'll probably have to wait about 2 seconds for the catchup scans, so it'll take far longer in practice.

We should have a metric for how many rangefeeds are using the scheduler or processor, so that operators can monitor progress. We can address that separately.